features/step_definitions/kata.rb

```diff
@@ -1,229 +1,103 @@
-      })
-      step %Q/the step should succeed/
-      # save the channel from subscription into cb.channel
-      step %Q/I extract the channel information from subscription and save it to the clipboard/
-      step %Q/I perform the :set_custom_channel_and_subscribe web action with:/, table(%{
-        | update_channel    | #{cb.channel} |
-        | install_mode      | OwnNamespace  |
-        | approval_strategy | Automatic     |
-      })
-      step %Q/the step should succeed/
-      step %Q/a pod becomes ready with labels:/, table(%{
-        | control-plane=controller-manager |
-      })
-    else
-      step %Q/I switch to cluster admin pseudo user/
-      step %Q|I obtain test data file "kata/release-#{cb.master_version}/deployment.yaml"|
-      @result = user.cli_exec(:apply, f: "deployment.yaml")
-      raise "Failed to deploy kata operator" unless @result[:success]
-      project(kata_ns)
-    end
+Then /^Kata runtime installed on selected worker nodes for #{QUOTED} kataconfig$/ do |kataconfig_name|
+  step %Q|I wait until number of completed kata runtime nodes for "#{kataconfig_name}" matches|
+end
 
-    # make sure kata-operator is running first before installing the kataconfig
-    step %Q/a pod becomes ready with labels:/, table(%{
-      | control-plane=controller-manager |
-    })
-    step %Q|I obtain test data file "kata/release-#{cb.master_version}/kataconfiguration_v1_kataconfig.yaml"|
-    @result = user.cli_exec(:apply, f: 'kataconfiguration_v1_kataconfig.yaml')
-    raise "Failed to apply kataconfig" unless @result[:success]
-    step %Q/I wait for "#{kata_config_name}" install to start/
-  else
-    logger.info("There's already an existing 'kataconfig' resuing it...")
-    project(kata_ns)
-    step %Q/I switch to cluster admin pseudo user/
-    # make sure kata-operator is running first before installing the kataconfig
-    step %Q/a pod becomes ready with labels:/, table(%{
-      | control-plane=controller-manager |
-    })
-  end
-  logger.info("Using kata image: #{pod.container_specs.first.image}")
-  step %Q/I wait until number of completed kata runtime nodes match for "#{kata_config_name}"/
+When /^I apply #{QUOTED} pod in #{QUOTED} namespace$/ do |pod_name, kata_ns|
+  step %Q|I obtain test data file "kata/#{pod_name}.yaml"|
+  @result = user.cli_exec(:apply, f: "#{pod_name}.yaml", n:kata_ns)
+  raise "Failed to deploy a pod" unless @result[:success]
 end
 
-Given /^I extract the channel information from subscription and save it to the#{OPT_SYM} clipboard$/ do |cb_name|
-  cb_name ||= :channel
-  step %Q/I store master major version in the :master_version clipboard/ if cb.master_version.nil?
-  step %Q|I obtain test data file "kata/release-#{cb.master_version}/subscription.yaml"|
-  channel = YAML.load(open('subscription.yaml')).dig('spec', 'channel')
-  logger.info("Subscription using channel: #{channel}")
-  cb[cb_name] = channel
+Then /^#{QUOTED} Pod should run using kata runtime$/ do |pod_name|
```

Comment:
@valtair1891 I don't think `Pod` should be cap here.

Reply:
done